### PR TITLE
Remove irrelevant mature UI toggle test

### LINKED ISSUE #2572: "e2e/search-query-server.spec.ts:94:9 › search query on SSR › url mature query is set, and can be unchecked using the Safer Browsing popup" is skipped
## Problem

"e2e/search-query-server.spec.ts:94:9 › search query on SSR › url mature query is set, and can be unchecked using the Safer Browsing popup" is skipped without any explanation or associated issue that I can find. When I unskip the test it does not pass locally.

## Description

Figure out why the test does not pass. If the test is a good test to have, fix it. Otherwise, if the test is not necessary, delete it from the test suite.

## Additional context

I've marked this critical as per our flaky test guidelines but because it's already been skipped for a long time, it doesn't exactly fall into the same category. But someone did write the test for a reason, so presumably it's a good one to have. It also only fails for two breakpoints rather than all of them, so the behaviour is verifiable in _some_ conditions. Perhaps it just needs to be wrapped in a conditional based on the breakpoint being tested and excluded from the suite for certain breakpoints.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #2572 by @sarayourfriend 

## Description

Removes an irrelevant mature UI toggle test. We will add additional test coverage for new sensntivity UI elements when they are added in #2368.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/test/playwright/e2e/search-query-server.spec.ts [hunk 1]
```diff
@@ -91,21 +91,13 @@ test.describe("search query on SSR", () => {
...
-      await page.click('button:has-text("Safer Browsing")')
-
-      const matureCheckbox = await page.locator("text=Show Mature Content")
-      await expect(matureCheckbox).toBeChecked()
-
-      await page.click("text=Show Mature Content")
-      await expect(page).toHaveURL("/search/image?q=cat")
-    })
+    test.fixme(
+      "URL sensntive query is set, and can be toggled using the UI.",
```
Comment (zackkrida) [inline frontend/test/playwright/e2e/search-query-server.spec.ts:95]:
```suggestion
      "URL sensitive query is set, and can be toggled using the UI.",
```

typo

File: frontend/test/playwright/e2e/search-query-server.spec.ts [hunk 2]
```diff
@@ -91,21 +91,13 @@ test.describe("search query on SSR", () => {
...
-
-      await page.click('button:has-text("Safer Browsing")')
-
-      const matureCheckbox = await page.locator("text=Show Mature Content")
-      await expect(matureCheckbox).toBeChecked()
-
-      await page.click("text=Show Mature Content")
-      await expect(page).toHaveURL("/search/image?q=cat")
-    })
+    test.fixme(
```
Comment (obulat) [inline frontend/test/playwright/e2e/search-query-server.spec.ts:94]: Now I kind of think we should just remove it. It doesn't really have any meaning now.

Comment (zackkrida) [reply frontend/test/playwright/e2e/search-query-server.spec.ts:94]: We could keep some of the implementation, if that is a more compelling reason to keep the test. I'm also fine with just deleting it.

**Comment by sarayourfriend** [comment]
I also agree with removing it. If any part of this is useful, we can easily retrieve it from version control. At this point there are more than sufficient references to it that multiple maintainers have seen to know where to find it. I have frankly very little confidence that a TODO that says "this test should be implemented as part of XYZ project" is more useful. On the other hand, I think the `.skip` here is good evidence that it is very easy for us to forget about a test like this.

My question is: what is the reason (value add) to keep this test, even a stub/reference for it _in the code_ without an associated feature? If we want to add todo tests for the feature, we can do so, but lets do it for the entire feature, rather than keeping a single test with a name referencing a legacy feature that may or may not have anything to do with how the new feature ends up working. To be the downside is that this has a high likelihood of being forgotten/ignored and creating confusion in the future when someone has to dig up our conversations about this again.

**Comment by zackkrida** [comment]
> If we want to add todo tests for the feature, we can do so, but lets do it for the entire feature, rather than keeping a single test with a name referencing a legacy feature that may or may not have anything to do with how the new feature ends up working. 

Agreed, I'll simply remove the code instead.

**Review by sarayourfriend — APPROVED** [review]
LGTM :+1: The PR description needs updating as well, just in case that wasn't on your radar :slightly_smiling_face:

**Review by AetherUnbound — APPROVED** [review]
Agreed with all the discussion here, removing sounds good!

**Pull request merged**